data-tck/src/main/groovy/io/micronaut/data/tck/tests/AbstractRepositorySpec.groovy

```diff
@@ -863,6 +877,108 @@ abstract class AbstractRepositorySpec extends Specification {
 
         expect:
         mealRepository.findById(meal.mid).get().currentBloodGlucose == 100
+
+        cleanup:
+        cleanupMeals()
+    }
+
+    void "test find one for update"() {
+        given:
+        def meal = mealRepository.save(new Meal(10))
+        def food = foodRepository.save(new Food("food", 80, 200, meal))
+
+        when:
+        def mealById = transactionManager.executeWrite { mealRepository.findByIdForUpdate(meal.mid) }
```

Comment:
I had to manually wrap the `find*ForUpdate` inside a write transaction since `find` methods execute in a read-only transaction, which do not allow `FOR UPDATE` queries. Write transactions could be used instead for these methods, although it doesn't make much sense to do a `SELECT FOR UPDATE` on its own. That's why I decided to keep it this way for the time being.